sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/documentanalysis/DocumentAnalysisClient.java

```diff
@@ -128,8 +153,34 @@ public final class DocumentAnalysisClient {
     public SyncPoller<OperationResult, AnalyzeResult>
         beginAnalyzeDocumentFromUrl(String modelId, String documentUrl,
                                     AnalyzeDocumentOptions analyzeDocumentOptions, Context context) {
-        return client.beginAnalyzeDocumentFromUrl(documentUrl, modelId,
-            analyzeDocumentOptions, context).getSyncPoller();
+        return beginAnalyzeDocumentFromUrlSync(documentUrl, modelId,
+            analyzeDocumentOptions, context);
+    }
+
+    SyncPoller<OperationResult, AnalyzeResult> beginAnalyzeDocumentFromUrlSync(String documentUrl, String modelId,
+        AnalyzeDocumentOptions analyzeDocumentOptions, Context context) {
+        if (CoreUtils.isNullOrEmpty(documentUrl)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'documentUrl' is required and cannot"
+                + " be null or empty"));
+        }
+        if (CoreUtils.isNullOrEmpty(modelId)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'modelId' is required and cannot"
+                + " be null or empty"));
+        }
+        final AnalyzeDocumentOptions finalAnalyzeDocumentOptions = getAnalyzeDocumentOptions(analyzeDocumentOptions);
+        context = enableSyncRestProxy(getTracingContext(context));
+        Context finalContext = context;
```

Comment:
why need this `finalContext`?

Reply:
variable needs to be final or effectively final before passing to lambda .